.github/workflows/runtests.yml

```
@@ -2,10 +2,42 @@ name: CI
 on: [pull_request]
 
 jobs:
+   cygwin:
+      timeout-minutes: 40
+      runs-on: windows-latest
```

I might not have got the syntax right, but I thin that adding this would stop you from having to manually configure the `shell` on each step.
```suggestion
      runs-on: windows-latest
      defaults:
        run:
          shell: C:\cygwin64\bin\bash.exe --login -o igncr '{0}'
```